api/v1beta2/ibmpowervscluster_webhook.go

```diff
@@ -102,11 +102,36 @@ func (r *IBMPowerVSCluster) validateIBMPowerVSClusterNetwork() *field.Error {
 	return nil
 }
 
+func (r *IBMPowerVSCluster) validateIBMPowerVSClusterLoadBalancers() (allErrs field.ErrorList) {
+	if err := r.validateIBMPowerVSClusterLoadBalancerNames(); err != nil {
+		allErrs = append(allErrs, err...)
+	}
+
+	isPublic := false
+	for _, loadbalancer := range r.Spec.LoadBalancers {
+		if *loadbalancer.Public {
+			isPublic = true
+			break
+		}
+	}
+
+	if !isPublic {
+		allErrs = append(allErrs, field.Invalid(field.NewPath("spec.LoadBalancers"), r.Spec.LoadBalancers, "Expect atleast one of the load balancer to be public"))
+	}
+
+	return allErrs
+}
+
 func (r *IBMPowerVSCluster) validateIBMPowerVSClusterLoadBalancerNames() (allErrs field.ErrorList) {
 	found := make(map[string]bool)
 	for i, loadbalancer := range r.Spec.LoadBalancers {
+		if loadbalancer.Name == "" {
+			continue
```

Comment:
It checks if LB with given ID is present and for the second time(with same ID), it will override the already set status and adds only one LB to the status field.